src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1751,6 +1752,10 @@ def forward(
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
         if labels is not None:
+            if labels.shape[1] > self.max_target_positions:
+                raise ValueError(
+                    f"Labels' sequence length {labels.shape[1]} cannot exceed the maximum allowed length of {self.max_target_positions} tokens."
+                )
```

Comment:
This seems like something that could slow down the forward pass during training; it seems to me like that could/should be handled from the processor side of things, but I understand `labels` aren't managed by the processor. 

WDYT @ylacombe?

Reply:
I'm not aware of this kind of verification being done by the processor or the tokenizer tbh! 
Are the costs of computing the shape and doing an `if` statement high enough to actually slow down training ? I'm not that sure

Happy to take other opinions here, maybe @eustlb or @ArthurZucker ?

Reply:
@ArthurZucker mentions that it's unlikely to slow down, so let's merge like this and eventually revert if we see a slowdown.